File: javascript/vulnerable-command-injection.js
const express = require("express");
const childProcess = require("child_process");

const app = express();

app.get("/diagnostics", (req, res) => {
  const host = req.query.host || "127.0.0.1";
  childProcess.exec(`ping -c 1 ${host}`, (error, stdout, stderr) => {
    if (error) {
      res.status(500).json({ error: stderr });
      return;
    }

    res.json({ output: stdout });
  });
});

app.listen(3000);

File: python/vulnerable_command_injection.py
import subprocess

from flask import Flask, request


app = Flask(__name__)


@app.get("/diagnostics")
def diagnostics():
    host = request.args.get("host", "127.0.0.1")
    output = subprocess.check_output(f"ping -c 1 {host}", shell=True, text=True)
    return {"output": output}


if __name__ == "__main__":
    app.run()
